src/backend/common/DepthStencilState.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BACKEND_COMMON_DEPTHSTENCILSTATE_H_
+#define BACKEND_COMMON_DEPTHSTENCILSTATE_H_
+
+#include "Forward.h"
+#include "Builder.h"
+#include "RefCounted.h"
+
+#include "nxt/nxtcpp.h"
+
+namespace backend {
+
+
+    class DepthStencilStateBase : public RefCounted {
+        public:
+            DepthStencilStateBase(DepthStencilStateBuilder* builder);
+
+            struct DepthInfo {
+                nxt::CompareFunction compareFunction = nxt::CompareFunction::Always;
+                bool depthWriteEnabled = false;
+            };
+
+            struct StencilFaceInfo {
+                nxt::CompareFunction compareFunction = nxt::CompareFunction::Always;
+                nxt::StencilOperation stencilFail = nxt::StencilOperation::Keep;
+                nxt::StencilOperation depthFail = nxt::StencilOperation::Keep;
+                nxt::StencilOperation depthStencilPass = nxt::StencilOperation::Keep;
+                uint32_t mask = 0xff;
+            };
+
+            struct StencilInfo {
+                bool stencilTestEnabled;
```

Comment:
stencilEnabled is never used.
Also the read and write mask will live inside StencilInfo